### PR TITLE
[MINOR][DOC] Fix window function docs

### PR DESCRIPTION
**Pull request by wecharyu**
### What changes were proposed in this pull request?
Minor change to fix the examples in `sql-ref-syntax-qry-select-window.md`.

### Why are the changes needed?
Doc fix.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Check locally.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: docs/sql-ref-syntax-qry-select-window.md [hunk 1]
```diff
@@ -94,107 +94,112 @@ INSERT INTO employees VALUES ("Paul", "Engineering", 29000, 23);
...
-| Jeff|  Marketing| 35000|   38|
-| Evan|      Sales| 32000|   38|
-| Lisa|      Sales| 10000|   35|
-| Alex|      Sales| 30000|   33|
-+-----+-----------+------+-----+
++-----+-----------+------+---+
+| name|       dept|salary|age|
++-----+-----------+------+---+
+|Chloe|Engineering| 23000| 25|
+| Paul|Engineering| 29000| 23|
```
Comment (allisonwang-db) [inline docs/sql-ref-syntax-qry-select-window.md:101]: Why are these changed?

File: docs/sql-ref-syntax-qry-select-window.md [hunk 2]
```diff
@@ -94,107 +94,112 @@ INSERT INTO employees VALUES ("Paul", "Engineering", 29000, 23);
...
 | Paul|Engineering| 29000|21000|
+| Jane|  Marketing| 29000|29000|
+| Jeff|  Marketing| 35000|29000|
+| Lisa|      Sales| 10000|10000|
+| Alex|      Sales| 30000|10000|
+| Evan|      Sales| 32000|10000|
 +-----+-----------+------+-----+
 
-SELECT name, salary,
+SELECT name, dept, salary,
```
Comment (allisonwang-db) [inline docs/sql-ref-syntax-qry-select-window.md:174]: good catch!

File: docs/sql-ref-syntax-qry-select-window.md (hunk 1 above)
Comment (wecharyu) [reply docs/sql-ref-syntax-qry-select-window.md]: Update the result with spark 4.0.0-preview1
